document/core/exec/instructions.rst

```diff
@@ -3016,26 +3016,104 @@ When the end of a :ref:`try <syntax-try>` instruction is reached without a jump,
 Throwing an exception with :ref:`tag address <syntax-tagaddr>` :math:`a`
 ........................................................................
 
+When an administrative throw occurs, then values, labels, |CAUGHTadm| instructions,
+and call frames are popped if necessary, until an appropriate exception handler is found
+on the top of the stack.
+
+ 1. Assert: due to validation, :math:`S.\STAGS[a]` exists.
+
+ 2. Let :math:`[t^n] \to []` be the :ref:`tag type <syntax-tagtype>` :math:`S.\STAGS[a].\TAGITYPE`.
+
+ 3. Assert: due to :ref:`validation <valid-throw>`, there are :math:`n` values on the top of the stack.
+
+ 4. Pop the :math:`n` values :math:`\val^n` from the stack.
+
+ 5. While the stack is not empty and the top of the stack is not an :ref:`exception handler <syntax-handler>`, do:
+
+    a. Pop the top element from the stack, prepending it to the :ref:`throw context <syntax-ctxt-throw>` of the exception: :math:`\XT[\val^n~(\THROWadm~a)]`.
+
+ 6. Assert: The stack is now either empty, or there is an exception handler on the top of the stack.
+
+ 7. If the stack is empty, then:
+
+    a. **TODO** *Return a result value representing the uncaught exception (will probably just be the same as 11.a.i. below).*
+
 .. todo::
-   Add prose for the following execution steps.
+   After PR #221 is resolved, this step should be filled in with a PR to specify uncaught exception results.
+
+8. Else there is an :ref:`exception handler <syntax-handler>` :math:`H` on the top of the stack.
+
+9. Pop the exception handler :math:`H` from the stack.
+
+10. Assert: :math:`H` is either of the form :math:`\CATCHadm\{a^?~\instr^\ast\}^k` or :math:`\DELEGATEadm\{l\}.`
+
+11. If :math:`H` is of the form :math:`\CATCHadm\{a^?~\instr^\ast\}^k`, then:
+
+    a. If :math:`k = 0`, then:
+
+       i. Push the throw context that we collected so far :math:`\XT[\val^n~(\THROWadm~a)]` onto the stack.
+
+    b. Else :math:`H` is of the form :math:`\CATCHadm\{a_1^?~\instr^\ast\}\{a'^?~\instr'^\ast\}^\ast`.
+
+    c. If :math:`a_1^? = \epsilon`, then:
+
+       i. Push :math:`\CAUGHTadm\{a~\val^n\}` onto the stack.
 
+       ii. Jump to the start of the instruction sequence :math:`\instr^\ast`.
+
+    d. Else if :math:`a_1 = a`, then:
+
+       i. Push :math:`\CAUGHTadm\{a~\val^n\}` onto the stack.
```

Comment:
> Does `val^n` mean a different thing from the line "ii" below? 

No, these `val^n` are the same `val^n` from the line "ii" below. So a `caught{ a val^n}` holds the exception while the instructions in the catch block is executed. Since this is a `catch x_a` (due to "d"), the values in the exception are pushed to the stack first.

> I guess this line doesn't mean actual values but just notes what the tag's type is... right?

In fact `caught` holds the actual exception, so its tag and its values.